### PR TITLE
Ellen Homework 1

### PR DESCRIPTION
**Pull request by Ellen-548**
Hey Arcade,

So I have no idea if I've done any of this right, but anyway...

After clicking lots of buttons and kind of fudging my way through, I managed to get the 'master' file thing open in VS Code. I then managed to add another 'branch' (which is called 'Ellen Homework 1'), and tried to do the task you had set me.

I wasn't quite sure what you meant about wrapping the whole introduction in a function, but instead I did write a function to greet the user before the bit about the cave?

Sorry if this is totally wrong. Please let me know if I need to do something else.

Thank you,
Ellen :)

### CODE REVIEW

File: Ellen Homework 1.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import time 
+import random 
+
+
+random_creatures = ["gorgon", "vampire", "gruffalo", "zombie", "minatour"]
+
+
+def greet_user(username):
+    print("Hello, " + username + "!")
+
```
Comment (arcadecoder) [inline Ellen Homework 1.py:21]: This is great! You can do the same with the intro just by taking all the print statements & conditionals we used in the last session and putting it in a function - I've done an example here: https://gist.github.com/arcadecoder/4486c0500fe288619fdaa1bf0d069111 

Then you can call the function using `intro()`

File: Ellen Homework 1.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+random_creatures = ["gorgon", "vampire", "gruffalo", "zombie", "minatour"]
+
+
+def greet_user(username):
+    print("Hello, " + username + "!")
+    
+greet_user('arcade')
+greet_user('ellen')
+greet_user('sophie')
+
```
Comment (arcadecoder) [inline Ellen Homework 1.py]: If you wanted you could add this function inside the newly created intro function.

**Review by arcadecoder — CHANGES_REQUESTED** [review]
Amazing that you created the pull request and figured it out! Super impressed, I've added some comments in the code so you can try changing it and then do another push to the same branch & I'll review again.

**Comment by Ellen-548** [comment]
> Amazing that you created the pull request and figured it out! Super impressed, I've added some comments in the code so you can try changing it and then do another push to the same branch & I'll review again.

Thank you so much!! Going to go through your comments and will re-upload in a bit :)

File: Ellen Homework 1.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+random_creatures = ["gorgon", "vampire", "gruffalo", "zombie", "minatour"]
+
+
+def intro():
+
+    def greet_user(username):
+        print("Hello, " + username + "!")
+
```
Comment (arcadecoder) [inline Ellen Homework 1.py:12]: The function definition of greet user needs to be above you creating the def intro() function. Generally you don't need to define a function inside another unless there are specific use cases.

**Review by arcadecoder — APPROVED** [review]

**Pull request merged**